lib/filters/http-headers.js

```diff
@@ -20,6 +20,11 @@ function httpHeaders (obj) {
 
 function filterSensitiveHeaders (headers) {
   for (const key in headers) {
+    // undefined headers will be dropped when serialized as
+    // json, and don't need to be redacted
+    if (typeof headers[key] === 'undefined') {
```

Comment:
nit: This works as well and theoretically should require less work from the JS engine:

```suggestion
    if (headers[key] === undefined) {
```